packages/web-client/app/utils/web3-strategies/layer-1-connection-manager.ts

```diff
@@ -109,7 +118,12 @@ export class ConnectionManager {
   }
 
   createStrategy(providerId: WalletProviderId) {
-    if (providerId === 'metamask') {
+    if (config.environment == 'test') {
```

Comment:
When you see this pattern in a class under test: "if test env then instantiate this 'fake' collaborator, else instantiate the 'real' collaborator", it is a good sign that you should make this collaborator a constructor argument rather than have the class under test be responsible for instantiating it.

Reply:
I don't think we can make this collaborator (`ConnectionStrategy` subclasses) a constructor argument of `ConnectionManager` since `ConnectionManager` is long-lived and we can't tell which strategy will be used until the moment of connection/reconnection though.

Reply:
You could inject a factory in that case.